spec/ttdrest/concerns/ad_group_spec.rb

```diff
@@ -0,0 +1,596 @@
+              "CurrencyCode": "USD"
+            }
+          )
+        end
+
+        context 'when nil' do
+          let(:max_bid_cpm) { nil }
+
+          it 'does not contain the key at all' do
+            expect(
+              client.build_ad_group_data(ad_group_id, campaign_id, name, budget_settings, base_bid_cpm, max_bid_cpm, creative_ids, params)['RTBAttributes'].keys
+            ).to_not include("MaxBidCPM")
+          end
+        end
+      end
+
+      describe 'creative_ids' do
+        it 'determines the value of RTBAttributes["CreativeIds"]' do
+          expect(
+            client.build_ad_group_data(ad_group_id, campaign_id, name, budget_settings, base_bid_cpm, max_bid_cpm, creative_ids, params)['RTBAttributes']
+          ).to include(
+            "CreativeIds" => [1,2,3] 
+          )
+        end
+        context 'when empty' do
+          let(:creative_ids) { [] }
+          it 'does not contain the key at all' do
+            expect(
+              client.build_ad_group_data(ad_group_id, campaign_id, name, budget_settings, base_bid_cpm, max_bid_cpm, creative_ids, params)['RTBAttributes'].keys
+            ).to_not include("CreativeIds")
+          end
+        end
+      end
+
+      describe 'params hash' do
+        describe 'description' do
+          it 'determines the value of Description' do
+            expect(
+              client.build_ad_group_data(ad_group_id, campaign_id, name, budget_settings, base_bid_cpm, max_bid_cpm, creative_ids, params)
+            ).to include({ "Description" => 'foo' })
+          end
+
+          context 'when nil' do
+            let(:description) { nil }
+
+            it 'does not contain the key at all' do
+              expect(
+                client.build_ad_group_data(ad_group_id, campaign_id, name, budget_settings, base_bid_cpm, max_bid_cpm, creative_ids, params).keys
+              ).to_not include("Description")
+            end
+          end
+        end
+
+        describe 'is_enabled' do
+          it 'determines the value of IsEnabled' do
+            expect(
+              client.build_ad_group_data(ad_group_id, campaign_id, name, budget_settings, base_bid_cpm, max_bid_cpm, creative_ids, params)
+            ).to include({ "IsEnabled" => true })
+          end
+
+          context 'when nil' do
+            let(:is_enabled) { nil }
+
+            it 'does not contain the key at all' do
+              expect(
+                client.build_ad_group_data(ad_group_id, campaign_id, name, budget_settings, base_bid_cpm, max_bid_cpm, creative_ids, params).keys
+              ).to_not include("IsEnabled")
+            end
+          end
+        end
+
+        describe 'industry_category_id' do
+          it 'determines the value of IndustryCategoryId' do
+            expect(
+              client.build_ad_group_data(ad_group_id, campaign_id, name, budget_settings, base_bid_cpm, max_bid_cpm, creative_ids, params)
+            ).to include({ "IndustryCategoryId" => 54 })
```

Comment:
magic number

Reply:
is this the name of a code smell? 42 is what i think when i see magic numbers, but this was also the example value in the ttd api docs 😬

Reply:
Yeah, magic number is the code smell for "here's a raw value, but hard to tell WHY it should be here.

relevant fix here would be `            ).to include({ "IndustryCategoryId" => industry_category_id })`